src/runtime/ExtendedNativeFunctionObject.h

```diff
@@ -0,0 +1,27 @@
+#ifndef __EscargotExtendedNativeFunctionObject__
+#define __EscargotExtendedNativeFunctionObject__
+
+#include "NativeFunctionObject.h"
+
+namespace Escargot {
+
+class ExtendedNativeFunctionObject : public NativeFunctionObject {
+public:
+    ExtendedNativeFunctionObject(ExecutionState& state, NativeFunctionInfo info, SmallValueVector values);
+
+    virtual bool isExtendedNativeFunctionObject() const
+    {
+        return true;
+    }
+
+    SmallValueVector getInternalValues()
+    {
+        return m_values;
+    }
```

Comment:
How about clearly returning the reference of the SmallValueVector instead of copying it?